util/index.ts

```diff
@@ -22,3 +23,15 @@ export const getChain = (chainId: number) => {
   }
   throw new Error(`no chain found at chainId ${chainId}`);
 };
+
+export const getBlockExplorerUrl = (txHashOrAddress: string, chainId: number) => {
+  const group = isHex(txHashOrAddress) ? (txHashOrAddress.length === 42 ? 'address' : 'tx') : 'ens';
+  const chain = getChain(chainId);
+  const networkPrefix =
```

Comment:
If the `blockExplorers` property doesn't exist, it's an antipattern to default to https://etherscan.io, as that would be an invalid link. Instead, we should revert. Can simplify to:

```ts
if(!chain.blockExplorers) throw new Error("No block explorer found.")
return `${networkPrefix}/tx/${txHash}`
```